### ISSUE TITLE
Error Building on OS X 10.11.6

### ISSUE THREAD

**Issue by dbl001** [plaidml/plaidml#60]
I'm trying to build 'plaidml' on a Mac running OS X. 10.11.6.
I'm getting the following error:

ERROR: /Users/davidlaxer/plaidml/tile/lang/BUILD:141:1: Executing genrule //tile/lang:grammar failed (Exit 1).
/bin/bash: /usr/local/opt/bison/bin/bison: No such file or directory

$ bazel version
Build label: 0.7.0
Build target: bazel-out/darwin_x86_64-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Wed Oct 18 14:25:46 2017 (1508336746)
Build timestamp: 1508336746
Build timestamp as int: 1508336746

$ bazel build -c opt plaidml:wheel plaidml/keras:wheel

...............
WARNING: The lower priority option '-c opt' does not override the previous value '-c opt'.
INFO: Found 2 targets...
INFO: From BuildWheel plaidml/keras/wheel.pkg/dist/plaidml_keras-0.0.0.dev0-py2.py3-none-any.whl:
running bdist_wheel
running build
running build_py
creating build
creating build/lib
creating build/lib/plaidml
creating build/lib/plaidml/keras
copying pkg/plaidml/keras/__init__.py -> build/lib/plaidml/keras
copying pkg/plaidml/keras/backend.py -> build/lib/plaidml/keras
installing to build/bdist.macosx-10.7-x86_64/wheel
running install
running install_lib
creating build/bdist.macosx-10.7-x86_64
creating build/bdist.macosx-10.7-x86_64/wheel
creating build/bdist.macosx-10.7-x86_64/wheel/plaidml
creating build/bdist.macosx-10.7-x86_64/wheel/plaidml/keras
copying build/lib/plaidml/keras/__init__.py -> build/bdist.macosx-10.7-x86_64/wheel/plaidml/keras
copying build/lib/plaidml/keras/backend.py -> build/bdist.macosx-10.7-x86_64/wheel/plaidml/keras
running install_egg_info
running egg_info
creating pkg/plaidml_keras.egg-info
writing requirements to pkg/plaidml_keras.egg-info/requires.txt
writing pkg/plaidml_keras.egg-info/PKG-INFO
writing top-level names to pkg/plaidml_keras.egg-info/top_level.txt
writing dependency_links to pkg/plaidml_keras.egg-info/dependency_links.txt
writing entry points to pkg/plaidml_keras.egg-info/entry_points.txt
writing manifest file 'pkg/plaidml_keras.egg-info/SOURCES.txt'
reading manifest file 'pkg/plaidml_keras.egg-info/SOURCES.txt'
writing manifest file 'pkg/plaidml_keras.egg-info/SOURCES.txt'
Copying pkg/plaidml_keras.egg-info to build/bdist.macosx-10.7-x86_64/wheel/plaidml_keras-0.0.0.dev0-py2.7.egg-info
running install_scripts
creating build/bdist.macosx-10.7-x86_64/wheel/plaidml_keras-0.0.0.dev0.dist-info/WHEEL
INFO: From Compiling external/protobuf/src/google/protobuf/compiler/js/embed.cc [for host]:
external/protobuf/src/google/protobuf/compiler/js/embed.cc:37:12: warning: unused variable 'output_file' [-Wunused-const-variable]
const char output_file[] = "well_known_types_embed.cc";
           ^
1 warning generated.
ERROR: /Users/davidlaxer/plaidml/tile/lang/BUILD:141:1: Executing genrule //tile/lang:grammar failed (Exit 1).
/bin/bash: /usr/local/opt/bison/bin/bison: No such file or directory
INFO: Elapsed time: 359.894s, Critical Path: 10.72s
David-Laxers-MacBook-Pro:plaidml davidlaxer$ 
David-Laxers-MacBook-Pro:plaidml davidlaxer$ sudo port install bison
Password:
--->  Computing dependencies for bison
--->  Cleaning bison
--->  Scanning binaries for linking errors
--->  No broken files found.                             
David-Laxers-MacBook-Pro:plaidml davidlaxer$ bison
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bison: missing operand after `/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bison'
Try `/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bison --help' for more information.
David-Laxers-MacBook-Pro:plaidml davidlaxer$ bazel build -c opt plaidml:wheel plaidml/keras:wheel
WARNING: The lower priority option '-c opt' does not override the previous value '-c opt'.
INFO: Found 2 targets...
ERROR: /Users/davidlaxer/plaidml/tile/lang/BUILD:141:1: Executing genrule //tile/lang:grammar failed (Exit 1).
/bin/bash: /usr/local/opt/bison/bin/bison: No such file or directory
INFO: Elapsed time: 28.168s, Critical Path: 8.71s

$ which bison
/usr/bin/bison

$ bison --version
bison (GNU Bison) 2.3
Written by Robert Corbett and Richard Stallman.

**Comment by jbruestle** [comment]
On Mac OS, the apple supplied version of bison and flex are quite out of date (current bison version is 3.0 for example) and won't properly build tile.y and tile.l.  Thus, we recommend installing the homebrew versions of bison and flex (https://brew.sh), although you can also install them from source.  Since we presume homebrew versions, we use '/usr/local/opt/bison/bin/bison' as the path for bison for example.  If you do install the latest version elsewhere, see the rule in bzl/plaidml.bzl to adjust paths as needed.  Besides bison, flex, and of course Bazel itself, the rest of the Apple provided OS/X toolchain should be sufficient.  Although that said, if you just want the current release, all you really need to do is 'pip install plaidml-keras', which is generally sufficient for developing networks in Keras or writing Tile code, and our recommended way to get started.  I'm going to close this ticket as answered, but feel free to reopen if you run into further problems.

**Comment by dbl001** [comment]
I’m still using MacPorts.  After updating plaidml.bzl to reference the ‘ports’ version of bison / flex (as opposed to the Apple version), 
plaidml built successfully.

Thanks!

**Comment by brianretford** [comment]
I'll add these notes to a BUILDING file

**Comment by brianretford** [comment]
Fixed in #62
